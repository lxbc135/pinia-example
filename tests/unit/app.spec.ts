// Example using real Pinia store
import { createPinia, setActivePinia, Pinia } from 'pinia'
import { createApp } from 'vue';
import { mount } from '@vue/test-utils'
import { useTodosStore } from '../../src/stores/useTodosStore';
import App from '../../src/App.vue';

describe('App', () => {
  let pinia: Pinia;

  beforeEach(() => {
    const app = createApp({});
    pinia = createPinia();
    app.use(pinia);
    setActivePinia(pinia);
  });

  it('should init todos from store', () => {
    const store = useTodosStore();
    store.addTodo('todo1');
    expect(store.todos.length).toBe(1);
    const wrapper = mount(App, {
      global: {
        plugins: [pinia], // use real pinia store
      },
    });
    const activeTodos = wrapper.findAll('li');
    expect(activeTodos.length).toBe(1);
    expect(activeTodos.at(0)?.text()).toContain('todo1');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should add todo', async () => {
    const wrapper = mount(App, {
      global: {
        plugins: [pinia],
      },
    });
    const input = wrapper.find('#todo');
    await input.setValue('todo1');
    await wrapper.find('form').trigger('submit');
    await wrapper.vm.$nextTick(); // required to see effect
    const todos = wrapper.findAll('li');
    expect(todos.length).toBe(1);
    expect(todos[0].text()).toContain('todo1');
  });
});
