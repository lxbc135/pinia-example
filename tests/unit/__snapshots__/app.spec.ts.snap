// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should init todos from store 1`] = `
"<div class=\\"p-5\\">
  <h1 class=\\"text-5xl text-center\\">Pinia Todo Example</h1>
  <form class=\\"flex justify-center\\"><input type=\\"text\\" id=\\"todo\\" class=\\"bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block p-2.5 w-full max-w-lg my-10\\" placeholder=\\"My task is...\\" required=\\"\\"></form>
  <div class=\\"my-10 max-w-lg mx-auto\\">
    <h2 class=\\"text-2xl text-gray-900 font-bold mb-2\\">Active Todos</h2>
    <ul>
      <li class=\\"flex items-center justify-between p-2 border-2 border-solid border-gray-300 rounded-lg my-1\\"><span class=\\"text-gray-900 text-lg break-all\\">todo1</span>
        <div class=\\"ml-2 flex gap-1\\"><button class=\\"p-2 border-solid border-2 border-gray-400 rounded-lg text-gray-500 hover:text-green-500 hover:border-green-500\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"h-6 w-6\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" stroke-width=\\"2\\">
              <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" d=\\"M5 13l4 4L19 7\\"></path>
            </svg></button><button class=\\"p-2 border-solid border-2 border-gray-400 rounded-lg text-gray-500 hover:text-red-500 hover:border-red-500\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"h-6 w-6\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" stroke-width=\\"2\\">
              <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" d=\\"M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16\\"></path>
            </svg></button></div>
      </li>
    </ul>
  </div>
  <!--v-if--><button type=\\"button\\" title=\\"Clear locally stored todos\\" class=\\"transition-colors ease-out duration-300 absolute right-5 bottom-5 p-3 border-2 border-solid border-gray-500 text-gray-500 rounded-lg hover:bg-gray-700 hover:border-gray-700 hover:text-white\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"h-6 w-6\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" stroke-width=\\"2\\">
      <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" d=\\"M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16\\"></path>
    </svg></button>
</div>"
`;
